src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -367,3 +368,85 @@ def _try_fetch(self) -> Iterable[MigrationStatus]:
     def _iter_schemas(self):
         for catalog in self._ws.catalogs.list():
             yield from self._ws.schemas.list(catalog_name=catalog.name)
+
+
+class TablesInMountsMigrator:
+    def __init__(
+        self,
+        tables_in_mounts: TablesInMounts,
+        ws: WorkspaceClient,
+        backend: SqlBackend,
+        table_mapping: TableMapping,
+    ):
+        self._backend = backend
+        self._ws = ws
+        self._tm = table_mapping
+        self._tim = tables_in_mounts
+
+    def create_tables_in_uc(self):
```

Comment:
What if there is already a HMS table on top on the same location crawled by `TablesInMounts`? Is it possible that the `tables` table in inventory database has two items that pointing to a same location? For example something like:
1. hive_metastore, schema, hms_table, EXTERNAL, s3://bucket/table_delta
2. hive_metastore, mounted_bucket, table_delta, EXTERNAL, s3://bucket/table_delta

Reply:
In this case, the execution of `sql_migrate_table_in_mount` will fail. You should capture the exception and skip the migration if the failure is table already exists or location overlap.

Reply:
In this case, the execution of `sql_migrate_table_in_mount` will fail. We should capture the failure and skip the migration if the failure is due to table exists or location overlap.

Reply:
@william-conti Maybe we can filter out the table in mounts by skipping the `loaded_records` that have location already presented in `cached_results` in the `_snapshot` function, to solve this conflict issue.